emitter.py

```diff
@@ -40,10 +67,22 @@ def http_emitter(message, log, agentConfig, endpoint):
 
     # Post back the data
     try:
-        payload = json.dumps(message)
-    except UnicodeDecodeError:
-        message = remove_control_chars(message)
-        payload = json.dumps(message)
+        try:
+            payload = json.dumps(message)
+        except UnicodeDecodeError:
+            newmessage = remove_control_chars_from(message, log)
+            payload = json.dumps(newmessage)
+    except UnicodeDecodeError as ude:
+        log.error('http_emitter: Unable to convert message to json %s ' % str(ude))
```

Comment:
What do you think of `log.exception('http_emitter: Unable to convert message to json %s', ude)` - https://www.simonmweber.com/2014/11/24/python-logging-traps.html ?